exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -302,14 +302,13 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		}
 
 		// Since maps are not ordered, we need to order them to get the same hash at each reconcile.
-		keys := make([]string, 0)
-		data, ok := unstructuredObj.UnstructuredContent()["data"]
-		if !ok {
-			errList = append(errList, errors.New("failed to get data field from the resource"))
+		unstructuredData, _, err := unstructured.NestedMap(unstructuredObj.UnstructuredContent(), "data")
```

Comment:
Not sure if this is changing behaviour intentionally but to keep current behaviour this should also consider the bool returned:

```
unstructuredData, found, err := unstructured.NestedMap(unstructuredObj.UnstructuredContent(), "data")
if err != nil || !found {...}
```

Reply:
I'm not considering found because as far as I know having a ConfigMap or a Secret without data is a valid use case (we also have it in out tests).